src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2441,24 +2438,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
       return false;
     case Op_VectorLoadShuffle:
     case Op_VectorRearrange:
-      if (vlen < 4) {
-        return false;
-      }
-      break;
-    // Some types of VectorCast are not implemented for now.
-    case Op_VectorCastI2X:
-      if (bt == T_BYTE) {
-        return false;
-      }
-      break;
+    // We're not going to have VectorCast with bit_size less than 64 bits
+    // support except for byte vector with vlen more than or equal to 4.
+    case Op_VectorCastB2X:
     case Op_VectorCastS2X:
-      if (vlen < 4 || bit_size < 64) {
-        return false;
-      }
-      break;
-    case Op_VectorCastF2X:
-    case Op_VectorCastD2X:
-      if (bt == T_INT || bt == T_SHORT || bt == T_BYTE || bt == T_LONG) {
+      if (vlen < 4) {
```

Comment:
I suggested to test this PR with the latest code. As some vector sizes which should not be supported for `VectorReinterpret` and `VectorCast*2X` have been fixed after https://github.com/openjdk/jdk/pull/5160. E.g. 128short => 64int.

For the case above, as the species are different, it
1. reinterprets 8 short to 2 short
2. casts 2 short to 2 int

Since we don't support short type with element size less than 4, this situation should be detected as unsupported when trying to generate `VectorReinterpret` node with 2 short, which in current branch is mistaken for 4 short.

I think those workaround code for jdk17(https://github.com/openjdk/jdk17/compare/master...theRealELiu:JDK-8268966) could be removed entirely after this work.

Reply:
You are right, I'll fix it.

Reply:
Kindly remind that those workaround code should be removed after merging.